src/stores/recentBlocks.js

```diff
@@ -143,160 +127,40 @@ export const useRecentBlocksStore = defineStore('recentBlocks', () => {
 
   /* HANDLING COMMUNICATION OVER WEBSOCKET */
 
-  function processSocketMessage(message) {
-    messageBuffer.value.push(message)
-
-    if (messageBuffer.value.length > MESSAGE_BUFFER_SIZE_LIMIT) {
-      return resetMessageBuffer()
-    }
-
-    for (let i = messageBuffer.value.length - 1; i >= 0; i--) {
-      if (!consumeBufferMessage()) {
-        break
-      }
-    }
-  }
-
-  function consumeBufferMessage() {
-    for (let i = 0; i < messageBuffer.value.length; i++) {
-      const message = messageBuffer.value[i]
-      let success = false
-
-      switch (message.subscription) {
-      case 'KeyBlocks':
-        success = processKeyblockUpdate(message.payload)
-        break
-      case 'MicroBlocks':
-        success = processMicroblockUpdate(message.payload)
-        break
-      default:
-        success = processTransactionUpdate(message.payload)
-        break
+  async function processSocketMessage(message) {
+    switch (message.subscription) {
+    case 'KeyBlocks':
+      fetchDeltaStats()
+      break
+    case 'MicroBlocks':
+      fetchTotalTransactionsCount()
+      await fetchKeyblocks()
+
+      try {
+        await fetchSelectedKeyblockMicroblocks(selectedKeyblock.value.hash)
+
+        if (isFirstMicroblockSelected.value && isFirstKeyblockSelected.value) {
+          await fetchSelectedMicroblockTransactions()
+        }
+      } catch (error) {
+        // ignore 400 errors when fetching data by a non-existing microblock
+        // as they are caused by microforks
+        if (error?.response.status !== 400) {
+          throw error
+        }
       }
 
-      if (success) {
-        messageBuffer.value.splice(i, 1)
-        return true
+      // sometimes delta stats are not yet available on keyblock message, so retry fetching them again
+      if (selectedDeltaStats.value === null) {
+        await fetchDeltaStats()
```

Comment:
Is this a temporary solution (mdw issue)? If yes, is there a task to improve it?